extension-style/src/main/java/com/mapbox/maps/extension/style/layers/LayerExt.kt

```diff
@@ -18,23 +19,30 @@ import com.mapbox.maps.logE
  * @return StyleLayerPlugin
  */
 fun StyleManagerInterface.getLayer(layerId: String): Layer? {
-  return this.getStyleLayerProperty(layerId, "type").silentUnwrap<String>()?.let { type ->
-    when (type) {
-      "background" -> BackgroundLayer(layerId).also { it.delegate = this }
-      "location-indicator" -> LocationIndicatorLayer(layerId).also { it.delegate = this }
-      "sky" -> SkyLayer(layerId).also { it.delegate = this }
-      "circle" -> CircleLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "fill-extrusion" -> FillExtrusionLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "fill" -> FillLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "heatmap" -> HeatmapLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "hillshade" -> HillshadeLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "line" -> LineLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "raster" -> RasterLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "symbol" -> SymbolLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      else -> {
-        logE(TAG, "Layer type: $type unknown.")
-        null
-      }
+  fun source(): String = getStyleLayerProperty(layerId, "source").unwrap()
+  val type = getStyleLayerProperty(layerId, "type").silentUnwrap<String>()
+  return when (type) {
+    "background" -> BackgroundLayer(layerId)
+    "location-indicator" -> LocationIndicatorLayer(layerId)
+    "sky" -> SkyLayer(layerId)
+    "circle" -> CircleLayer(layerId, source())
+    "fill-extrusion" -> FillExtrusionLayer(layerId, source())
+    "fill" -> FillLayer(layerId, source())
+    "heatmap" -> HeatmapLayer(layerId, source())
+    "hillshade" -> HillshadeLayer(layerId, source())
+    "line" -> LineLayer(layerId, source())
+    "raster" -> RasterLayer(layerId, source())
+    "symbol" -> SymbolLayer(layerId, source())
+    else -> {
+      logE(TAG, "Layer type: $type unknown.")
+      null
+    }
+  }?.also { result ->
+    result.delegate = this
+    try {
+      result.cachedLayerPropertiesValue = getStyleLayerProperties(layerId).value
+    } catch (e: ClassCastException) {
+      // Exception can be thrown during unit tests (impossible to mock getStyleLayerProperties)
```

Comment:
why couldn't that be mocked? can you provide a bit more context?

Reply:
`getStyleLayerProperties` returns `Expected<String, Value>`, but was unable to mock getStyleLayerProperties to return a valid Expected object since Expected constructor is not exposed.  Maybe I misunderstood something, but at least trying the following does not compile: 
`every { style.getStyleLayerProperties("id") } returns Expected(null, null)
`

But now I noticed we have this in some tests:
`private val valueExpected = mockk<Expected<String, Value>>(relaxUnitFun = true, relaxed = true)
`

and then it would be possible to do:
`every { style.getStyleLayerProperties("id") } returns valueExpected
`

So I could remove this try catch and add this each of the Layer tests.

Reply:
This is now fixed and the try-catch block has been removed